### PR TITLE
[Net] Rename RPC constants and annotation arguments.

### PR DESCRIPTION
**Pull request by Faless**
any -> any_peer
sync -> call_local
ordered -> unreliable_ordered

Multiplayer.RPC_MODE_ANY -> RPC_MODE_ANY_PEER
Multiplayer.TRANSFER_MODE_ORDERED -> TRANSFER_MODE_UNRELIABLE_ORDERED

As emerged from the feedback to my [blog post](https://godotengine.org/article/multiplayer-changes-godot-4-0-report-2) (which I'll need to update)

### CODE REVIEW

**Comment by akien-mga** [comment]
Should Mono attributes be updated again? #53106

**Comment by Faless** [comment]
Didn't know they were changed already, updated.

**Review by akien-mga — APPROVED** [review]

File: modules/gdscript/gdscript_parser.cpp [hunk 1]
```diff
@@ -3481,22 +3481,22 @@ bool GDScriptParser::network_annotations(const AnnotationNode *p_annotation, Nod
...
-			} else if (mode == "auth") {
+			if (mode == "any_peer") {
+				rpc_config.rpc_mode = Multiplayer::RPC_MODE_ANY_PEER;
+			} else if (mode == "authority") {
 				rpc_config.rpc_mode = Multiplayer::RPC_MODE_AUTHORITY;
-			} else if (mode == "sync") {
+			} else if (mode == "call_local") {
 				rpc_config.sync = true;
-			} else if (mode == "nosync") {
+			} else if (mode == "no_call_local") {
```
Comment (reduz) [inline modules/gdscript/gdscript_parser.cpp:3490]: maybe call_remote ? although I thought this was the default.

**Comment by reduz** [comment]
I think it looks good, my only concern in general is that the way unreliable_orered works is kind of misleading, but I guess it can be addressed in another PR.

File: modules/gdscript/gdscript_parser.cpp (hunk 1 above)
Comment (Faless) [reply modules/gdscript/gdscript_parser.cpp:3490]: yes, that is the default.

**Review by reduz — APPROVED** [review]

**Pull request merged**